app/models/vote.rb

```diff
@@ -0,0 +1,7 @@
+class Vote < ApplicationRecord
+  belongs_to :user
+  belongs_to :work, counter_cache: true
+
+  validates :work_id, uniqueness: { scope: :user_id,
+                                    message: "User cannot vote for the same work twice." }
```

Comment:
Good work getting this tricky `uniqueness` scope and the `counter_cache` figured out!